### ISSUE TITLE
Add Toggle Button to Show/Hide Menu with Close Icon

### ISSUE THREAD

**Issue by Vinay-Khanagavi** [streetcomplete/StreetComplete#5935]
### Feature Description:

Currently, the user can open the menu by clicking the button next to the hamburger menu icon, but there is no clear visual indication for closing the menu once it's opened. To enhance usability and improve the user experience, I propose adding a feature that toggles a close icon (X) when the menu is opened. the problem for example when user is already on surfaces and does not want to change there is no close button he need to click on the screen to remove the contents, sometimes if he click somewhere where did not wanted to click it get's annoying 

- When the user clicks the same button, the menu should close, and the icon should revert back to its original state.

- We can use MUI to implement this feature:
`import CloseIcon from '@mui/icons-material/Close';`

**Screenshots:**
![Screenshot 2024-09-30 004525](https://github.com/user-attachments/assets/fa4defc2-ba0f-42e9-b0a4-a34d7ace8e3b) ![Screenshot 2024-09-30 003421](https://github.com/user-attachments/assets/1a758589-9d0c-406e-a0df-6931dfd2bdb5)  

**_Why It’s Needed:_**

- Improves Usability: A clear, intuitive visual cue (the "X" close icon) helps users understand how to close the menu easily after it’s opened.

- Consistency: It aligns with standard UI patterns where a toggle button shows both open and close states, providing a seamless interaction flow.

- Enhanced User Experience: This feature eliminates confusion, ensuring users can close the menu effortlessly with a single button click.

- Implementing this feature will significantly improve the navigation experience and make the interface more intuitive.

**Comment by mnalis** [comment]
**TL;DR:** I think it is fine as it is...

---

> Improves Usability: A clear, intuitive visual cue (the "X" close icon) helps users understand how to close the menu easily after it’s opened.

That makes absolute sense **_if you need to click on that icon_** to close. 

But AFAIK, you can click **_anywhere outside the window_** to close it if you want to back out. (In fact, that overlay icon is the only place you **should not**  click if you want to exit overlay window, as it will close and then immediately respawn the same window)

Note that **all other UI** of StreetComplete functions on the principle on **_"click outside the window to close it"_** - e.g. after clicking `+` in Things overlay, of after long pressing to open notes/track/otherapp submenu, or when clicking undo button, or when clicking main hamburger button, or when activating team mode, or when quest is open, or when you click on `Uh...` quest menu, or when you select `Can't say` answer, or when you are adding or updating a note, etc...

**You exit all of then by clicking outside of displayed window. (or pressing back button / gesture)**

So if that behaviour were to be changed, it would have to be changed _everywhere_ to be consistent. 

Given that current UI is superior for small screens (no extra space needed for close icon), simpler to implement (no special icons not handlers needed), more usable when using single hand (easier to reach any one of dozens regions, than a specific single button at specific position) more usable for "big thumbs" or people with fine motor control issues (bigger areas are easier to hit),  and also common in android UI (e.g. clicking outside of Android launcher group_of_app also closes it), I personally don't see the benefit of changing the UI.

> Implementing this feature will significantly improve the navigation experience and make the interface more intuitive.

There is a truism saying that _only_ intuitive thing is mother's tit, and _everything_ else is learned at some point later. :smiley_cat: 
I personally don't recall many android apps which have such _"`X` button at the place of original button"_; most employ other UIs. (in fact, such UI reminds me mostly of HTML5 mobile webpage hamburger menu, and extremely rare in native Android apps - at least the ones that I happen to use)

**Comment by Vinay-Khanagavi** [comment]
Yeah I think Adding a close button might not fit the current design, and I appreciate the explanation. 
keeping the UI simple and consistent is good cool too. 

I was just working on that but it's fine, There was one more think i wanted to ask

I've been exploring the current UI layout, and I noticed a couple of things that could be improved for a better user experience:
**Search Icon for "Download Data Here"**: I observed that the "Download data here" option has a search icon associated with it, which seems a bit confusing. Typically, users associate the search icon with finding or filtering content, whereas downloading data might be better represented by a download icon (⬇️). Could we consider updating the icon to something more intuitive, like a download symbol? 



![Screenshot 2024-09-30 005841](https://github.com/user-attachments/assets/1405853c-993f-41f0-96db-24e19e711a43)
**Alignment of the both icon :** The alignment of the menu items looks slightly off. It would improve user experience if we could align them consistently, ensuring equal spacing and proper centering within the menu box.

**_Thanks again! for the feedback_**
If you fine with this then things then I’ll close this issue.

**Comment by mnalis** [comment]
> I've been exploring the current UI layout, and I noticed a couple of things that could be improved for a better user experience:
> - **Search Icon** for "Download Data Here":

Might make sense to change that, thanks for suggestion! Originally it was called "Search for Quests here" (at least it is how it is still translated in some languages AFAICT), so that should at least explains _"why"_ of the magnifying glass.

> - **Alignment of the both icon** : The alignment of the menu items looks slightly off

Could you elaborate? Both _"Download data here"_ and _"Enter Team Mode"_ texts seem centered just fine to my untrained eye?

**Comment by Vinay-Khanagavi** [comment]
Okay sorry i didn't the meaning behind that thanks for explaining, and didn't mean by the text  "Download data here" and "Enter Team Mode" but the icon of it, don't you think they're too much left side

**Comment by mnalis** [comment]
> the icon of it, don't you think they're too much left side

Ah, now that you mention it, their icons _do_ seem _a little_ too close to the left edge. I would guess they are intentionally to the left side and not centered with the text, but maybe a **slightly bigger left margin would look better** (perhaps as much space as there it to the bottom of the screen?). One thing to consider is length of translated texts might be longer, so that (and reduced size of that window) might influence the outcome. 

I'd suggest trying how it looks with few different existing translations with different (i.e. extreme) lengths and suggesting an improvement with screenshot, if you'd like.

**Comment by westnordost** [comment]
This menu is a popup menu, also known as dropdown menu. This is a UI element that is not unique to StreetComplete.

Anyway, the suggestion for exchanging the icon makes sense to me. I will do it in the work-in-progress https://github.com/streetcomplete/StreetComplete/pull/5799 PR
